src/app/screens/SignTransaction/SignTransaction.tsx

```diff
@@ -33,48 +32,44 @@ export const SignTransaction = () => {
               uuid: currentAccount.uuid
             }
           : wallet,
-        tx: formattedTx
+        tx
       })
     );
   };
 
   const handleDeny = async () => {
-    if (currentTransaction) {
+    if (tx) {
       dispatch(denyCurrentTransaction());
       setError('');
     }
   };
 
   return (
     <>
-      {currentTransaction ? (
-        <pre>{JSON.stringify(currentTransaction, null, 2)}</pre>
-      ) : (
-        'Nothing to sign'
-      )}
+      {tx && <pre>{JSON.stringify(tx, null, 2)}</pre>}
       <br />
-      {currentAccount && currentAccount.type === WalletType.PRIVATE_KEY && (
+      {isWaiting && currentAccount && currentAccount.type === WalletType.PRIVATE_KEY && (
         <SignTransactionPrivateKey
           onAccept={handleAccept}
           onDeny={handleDeny}
-          tx={formattedTx}
+          tx={tx}
           currentAccount={currentAccount}
         />
       )}
-      {currentAccount && currentAccount.type === WalletType.MNEMONIC && (
+      {isWaiting && currentAccount && currentAccount.type === WalletType.MNEMONIC && (
         <SignTransactionMnemonic
           onAccept={handleAccept}
           onDeny={handleDeny}
-          tx={formattedTx}
+          tx={tx}
           currentAccount={currentAccount}
           setError={setError}
         />
       )}
-      {currentAccount && currentAccount.type === WalletType.KEYSTORE && (
+      {isWaiting && currentAccount && currentAccount.type === WalletType.KEYSTORE && (
         <SignTransactionKeystore
           onAccept={handleAccept}
           onDeny={handleDeny}
-          tx={formattedTx}
+          tx={tx}
           currentAccount={currentAccount}
         />
       )}
```

Comment:
We can simplify this by moving the components to a dictionary or function, since the props are the same across all components anyway.